### DISCUSSION TITLE
Output Score

### DISCUSSION THREAD

**Discussion by wperry1** [smacke/subsync#148] [Q&A] [answered]
Can you clarify the meaning of the score generated at the end of processing? The offset and framerate scale seem pretty clear but I can't seem to wrap my head around the score value.

Here's an example:
score: 199624.000
offset seconds: -0.090
framerate scale factor: 1.000

Thank you,
WP

**Comment by smacke** [comment, marked as answer]
It's kind of internal to ffsubsync. We define an objective and then try to find the alignment and scale factors that maximizes that objective. Roughly speaking we try to maximize the number of discretized segments that have speech both in the audio and in the subtitles. The number of total segments depends on the video length, and the proportion of those which are truly speech depends on the video, so it's not really interpretable across different pairs of reference / input subtitles. I mainly include it in the output so that I can try out different algorithms / heuristics to see if my alignments are improving or not, for a particular reference / input subtitle pair.

If you're interested here is where the sauce is made: https://github.com/smacke/ffsubsync/blob/e471020dec3192d3dc11a37029da84b934cb628e/ffsubsync/aligners.py#L47
